infer/src/istd/IStd.ml

```diff
@@ -5,10 +5,374 @@
  * LICENSE file in the root directory of this source tree.
  *)
 
-include Core_kernel
 
 [@@@warning "-32"]
 
+module Sys = struct
+  include Sys
+
+  let getenv var =
+    match getenv var with
+    | v -> Some v
+    | exception _ -> None
+
+  let stat_check_exn f ?(follow_symlinks=true) path =
+    let rec loop () =
+      try
+        f (if follow_symlinks then
+             Unix.LargeFile.stat path
+           else
+             Unix.LargeFile.lstat path)
+      with
+      | Unix.Unix_error(Unix.EINTR,_,_) -> loop ()
+      | Unix.Unix_error ((Unix.ENOENT|Unix.ENOTDIR), _, _) -> false
+    in
+    loop ()
+  ;;
+
+  let stat_check f ?(follow_symlinks=true) path =
+    try
+      if stat_check_exn ~follow_symlinks f path then
+        `Yes
+      else
+        `No
+    with
+    | Unix.Unix_error ((Unix.EACCES|Unix.ELOOP), _, _)   -> `Unknown
+  ;;
+
+  let file_exists     = stat_check     (fun _ -> true)
```

Comment:
Can you run ocamlformat? :) `make fmt` should sort it out for you.

Reply:
Sure :smile:  Do you want me to re-indent the existing commit and push -f, or add another commit?

Reply:
push -f sounds good.

Reply:
Done